src/main/java/io/percy/appium/providers/AppAutomate.java

```diff
@@ -76,23 +81,68 @@ public void executePercyScreenshotEnd(String name, String percyScreenshotUrl, St
         }
     }
 
-    public String screenshot(String name, String deviceName, Integer statusBarHeight, Integer navBarHeight,
-            String orientation, Boolean fullScreen) {
+    public String executePercyScreenshot(ScreenshotOptions options, Integer scaleFactor,
+        Integer deviceHeight) throws Exception {
+        try {
+            JSONObject arguments = new JSONObject();
+            JSONObject args = new JSONObject();
+            args.put("numOfTiles", options.getScreenLengths());
+            args.put("deviceHeight", deviceHeight);
+            arguments.put("state", "screenshot");
+            arguments.put("percyBuildId", System.getenv("PERCY_BUILD_ID"));
+            arguments.put("screenshotType", "fullpage");
+            arguments.put("scaleFactor", scaleFactor);
+            arguments.put("options", args);
+            JSONObject reqObject = new JSONObject();
+            reqObject.put("action", "percyScreenshot");
+            reqObject.put("arguments", arguments);
+            String resultString = driver
+                    .executeScript(String.format("browserstack_executor: %s", reqObject.toString())).toString();
+            JSONObject result = new JSONObject(resultString);
+            return result.get("result").toString();
+        } catch (Exception e) {
+            throw new Exception("Screenshot command failed");
+        }
+    }
+
+    public List<Tile> captureTiles(Boolean fullScreen, ScreenshotOptions options) throws Exception {
+        if (!options.getFullpageScreenshot()) {
+            return super.captureTiles(fullScreen, options);
+        }
+
+        Integer statusBar = getMetadata().statBarHeight();
+        Integer navBar = getMetadata().navBarHeight();
+        String response = executePercyScreenshot(options, getMetadata().scaleFactor(),
+            getMetadata().deviceScreenHeight());
+        JSONArray jsonarray = new JSONArray(response);
+        List<Tile> tiles = new ArrayList<Tile>();
+        for (int i = 0; i < jsonarray.length(); i++) {
+            JSONObject jsonobject = jsonarray.getJSONObject(i);
+            String sha = jsonobject.getString("sha").split("-")[0];
+            Integer headerHeight = jsonobject.getInt("header_height");
+            Integer footerHeight = jsonobject.getInt("footer_height");
+            tiles.add(new Tile(null, statusBar, navBar, headerHeight, footerHeight, fullScreen, sha));
+        }
+        return tiles;
+    }
+
+    public String screenshot(String name, ScreenshotOptions options, Boolean fullScreen) {
```

Comment:
This is good refactor, I would want to make same for c# as well. Btw you can add fullScreen to options as well.